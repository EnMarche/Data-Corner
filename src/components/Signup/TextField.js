import { TextField as MuiTextField } from '@mui/material'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'
import AlertBanner from 'ui/AlertBanner'

const TextInput = styled(MuiTextField)(
  ({ theme }) => `
  border-radius: 8px;
  border: none;
  color: ${theme.palette.blackCorner};
  background-color: ${theme.palette.gray100};
`
)

export const TextFieldFormik = ({ formik, label, inputProps = { maxLength: 255 }, ...props }) => (
  <>
    <TextInput
      {...props}
      error={!!formik.touched[label] && !!formik.errors[label]}
      fullWidth
      size="small"
      id={label}
      name={label}
      inputProps={inputProps}
      value={formik.values[label]}
      onChange={formik.handleChange}
    />
    {formik.touched[label] && formik.errors[label] && <AlertBanner severity="error" message={formik.errors[label]} />}
  </>
)

TextFieldFormik.propTypes = {
  formik: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  inputProps: PropTypes.object,
}

export const TextField = ({ value = '', inputProps = { maxLength: 255 }, onChange = () => {}, ...props }) => (
  <TextInput {...props} size="small" inputProps={inputProps} value={value} onChange={onChange} />
)

TextField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  inputProps: PropTypes.object,
}
