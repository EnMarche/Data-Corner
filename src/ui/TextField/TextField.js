import { TextField as TextFieldMui } from '@mui/material'
import { makeStyles } from '@mui/styles'
import PropTypes from 'prop-types'
import AlertBanner from 'ui/AlertBanner'

const useStyles = makeStyles(theme => ({
  textField: {
    border: `1px solid ${theme.palette.gray200}`,
    borderRadius: '8px',
    margin: '8px 0',
  },
}))

const TextField = ({ formik, label }) => {
  const classes = useStyles()

  return (
    <>
      <TextFieldMui
        fullWidth
        className={classes.textField}
        size="small"
        variant="outlined"
        multiline={label === 'body'}
        id={label}
        name={label}
        inputProps={{ maxLength: 255 }}
        value={formik.values[label]}
        onChange={formik.handleChange}
      />
      {formik.touched[label] && formik.errors[label] && <AlertBanner severity="error" message={formik.errors[label]} />}
    </>
  )
}

export default TextField

TextField.propTypes = {
  formik: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
}
