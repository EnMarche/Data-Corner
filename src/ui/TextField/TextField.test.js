import TextField from 'ui/TextField'
import { render } from '@testing-library/react'

jest.mock('@material-ui/core', () => ({
  TextField: ({ fullWidth, multiline, inputProps, ...props }) => <div className="mui-TextField-mock" {...props} />,
  makeStyles: () => () => ({ textField: 'textField' }),
}))

jest.mock('ui/AlertBanner', () => ({ children, ...rest }) => (
  <div className="AlertBannerMock" {...rest}>
    {children}
  </div>
))

describe('TextField', () => {
  const formik = {
    touched: {
      foo: 'foo-touched',
    },
    errors: {
      foo: 'foo-errors',
    },
    values: {
      foo: 'foo-values',
    },
  }

  it('displays TextField', () => {
    const { container } = render(<TextField label="foo" formik={formik} />)

    expect(container).toMatchSnapshot()
  })
})
