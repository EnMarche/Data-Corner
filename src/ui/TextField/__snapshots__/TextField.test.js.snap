// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField displays TextField 1`] = `
<div>
  <div
    class="css-1an706"
    error="false"
    id="foo"
    name="foo"
    value="foo-values"
    variant="outlined"
  />
  <div
    class="AlertBannerMock"
    message="foo-errors"
    severity="error"
  />
</div>
`;
